src/porepy/numerics/fv/mpfa.py

```diff
@@ -72,461 +69,260 @@ def discretize(self, g, data):
+            # been partially discretized even if it has not been their turn until now)
+            eliminate_face = np.where(
+                np.logical_not(np.in1d(l2g_faces, faces_in_subgrid))
+            )[0]
 
-        Parameters:
-            g (pp.Grid): grid to be discretized
-            k (pp.Second_order_tensor) permeability tensor. This is
-                the effective permeability; for lower-dimensional grids, any
-                aperture scaling of the permeability should be included in k.
-            bnd (pp.BoundaryCondition) class for boundary values
-            faces (np.ndarray) faces to be considered. Intended for partial
-                discretization, may change in the future
-            bnd (core.bc.bc) class for boundary values
-            deviation_from_plane_tol: The geometrical tolerance, used in the check to
-                rotate 2d grids
-            vector_source: discretization of the divergence of a vector source,
-                e.g. gravity in the flow equations. Default False. If True,
-                vector sources are discretized as part of the discrete flux operator
-            eta Location of pressure continuity point. Should be 1/3 for simplex
-                grids, 0 otherwise. On boundary faces with Dirichlet conditions,
-                eta=0 will be enforced.
-            eta_reconstruction Location of pressure reconstruction point on faces.
-            inverter (string) Block inverter to be used, either numba (default),
-                cython or python. See fvutils.invert_diagonal_blocks for details.
-            cells (np.array, int, optional): Index of cells on which to base the
-                subgrid computation. Defaults to None.
-            faces (np.array, int, optional): Index of faces on which to base the
-                subgrid computation. Defaults to None.
-            nodes (np.array, int, optional): Index of nodes on which to base the
-                subgrid computation. Defaults to None.
-
-            Note that if all of {cells, faces, nodes} are None, empty matrices will
-            be returned.
+            pp.fvutils.remove_nonlocal_contribution(eliminate_face, 1, *discr_fields)
 
-        Returns:
-            sps.csr_matrix (g.num_faces x g.num_cells): Flux discretization,
-                computed on a subgrid.
-            sps.csr_matrix (g,num_faces x g.num_faces): Boundary flux
-                discretization, computed on a subgrid
-            np.array (int): Global of the faces where the flux discretization is
-                computed.
+            # Split the discretization.
+            (
+                loc_flux,
+                loc_bound_flux,
+                loc_bound_pressure_cell,
+                loc_bound_pressure_face,
+                loc_vector_source,
+            ) = discr_fields
+
+            # Next, transfer discretization matrices from the local to the active grid
```

Comment:
What is "local" in this context?

Reply:
Refers to the subproblems (splitting due to max_memory).